### PR TITLE
Fix Users Issue due to capital names

### PR DESCRIPTION
**Pull request by mohityadav766**
### Describe your changes:

Fixes <issue-number>

#### UI side changes

- [x] lowercase email before sending it to `checkEmailInUse` endpoint
- [x] lowercase email while `login`
- [x] lowercase email when comparing it against token vs API response from `loggedInUser`

I worked on ... because ...

#
### Type of change:

- [ ] Bug fix
- [ ] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

File: openmetadata-service/src/main/java/org/openmetadata/service/security/CatalogPrincipal.java [hunk 1]
```diff
@@ -16,14 +16,14 @@
...
-  }
+@Getter
+public class CatalogPrincipal implements Principal {
+  private final String name;
+  private final String email;
 
-  public String getEmail() {
-    return email;
+  public CatalogPrincipal(String name, String email) {
+    this.name = name.toLowerCase();
```
Comment (chirag-madlani) [inline openmetadata-service/src/main/java/org/openmetadata/service/security/CatalogPrincipal.java:25]: Why do we want the name to be lowercase?

File: openmetadata-service/src/main/java/org/openmetadata/service/util/UserUtil.java [hunk 2]
```diff
@@ -330,4 +325,24 @@ public static boolean reSyncUserRolesFromToken(
 
     return syncUser;
   }
+
+  public static User getUser(String updatedBy, CreateUser create) {
+    return new User()
+        .withId(UUID.randomUUID())
+        .withName(create.getName().toLowerCase())
```
Comment (chirag-madlani) [inline openmetadata-service/src/main/java/org/openmetadata/service/util/UserUtil.java:332]: same

File: openmetadata-service/src/main/java/org/openmetadata/service/security/CatalogPrincipal.java (hunk 1 above)
Comment (mohityadav766) [reply openmetadata-service/src/main/java/org/openmetadata/service/security/CatalogPrincipal.java:25]: in postgres TestUser1 and testuser1 are both different, for us this is a same user

File: openmetadata-ui/src/main/resources/ui/src/components/Settings/Users/CreateUser/CreateUser.component.tsx [hunk 3]
```diff
@@ -123,7 +123,7 @@ const CreateUser = ({
       displayName: trim(displayName),
       roles: selectedRoles,
       teams: validTeam.length ? validTeam : undefined,
-      email: email,
+      email: lowerCase(email),
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/components/Settings/Users/CreateUser/CreateUser.component.tsx:126]: Let's use `toLower` to get the whole word a lowercase as `lowerCase` Converts string, as space-separated words, to lowercase.

File: openmetadata-ui/src/main/resources/ui/src/pages/SignUp/BasicSignup.component.tsx [hunk 4]
```diff
@@ -57,7 +57,7 @@ const BasicSignUp = () => {
       delete data['confirmPassword'];
     }
 
-    const request = data;
+    const request = { ...data, email: lowerCase(data.email) };
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/pages/SignUp/BasicSignup.component.tsx:60]: same here.

File: openmetadata-ui/src/main/resources/ui/src/utils/UserDataUtils.ts [hunk 5]
```diff
@@ -42,7 +42,7 @@ export const getUserDataFromOidc = (
 
   return {
     ...userData,
-    email,
+    email: lowerCase(email),
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/utils/UserDataUtils.ts:45]: same here.

**Review by Sachin-chaurasiya — COMMENTED** [review]
Let's add test for the changes.

**Review by chirag-madlani — APPROVED** [review]

**Review by ShaileshParmar11 — APPROVED** [review]

**Pull request merged**